src/java.base/share/classes/java/lang/runtime/TemplateRuntime.java

```diff
@@ -0,0 +1,236 @@
+                                             String... fragments) throws Throwable {
+        Objects.requireNonNull(lookup, "lookup is null");
+        Objects.requireNonNull(name, "name is null");
+        Objects.requireNonNull(type, "type is null");
+        Objects.requireNonNull(fragments, "fragments is null");
+
+        MethodHandle mh = StringTemplateImplFactory
+                .createStringTemplateImplMH(List.of(fragments), type).asType(type);
+
+        return new ConstantCallSite(mh);
+    }
+
+    /**
+     * String template bootstrap method for static final processors.
+     * The static arguments include the fragments list  and a {@link MethodHandle}
+     * to retrieve the value of the static final processor.
+     * The non-static arguments are the values.
+     *
+     * @param lookup          method lookup
+     * @param name            method name
+     * @param type            method type
+     * @param processorGetter {@link MethodHandle} to get static final processor
+     * @param fragments       fragments from string template
+     *
+     * @return {@link CallSite} to handle string template processing
+     *
+     * @throws NullPointerException if any of the arguments is null
+     * @throws Throwable            if linkage fails
+     */
+    public static CallSite processStringTemplate(MethodHandles.Lookup lookup,
+                                                 String name,
+                                                 MethodType type,
+                                                 MethodHandle processorGetter,
+                                                 String... fragments) throws Throwable {
+        Objects.requireNonNull(lookup, "lookup is null");
+        Objects.requireNonNull(name, "name is null");
+        Objects.requireNonNull(type, "type is null");
+        Objects.requireNonNull(processorGetter, "processorGetter is null");
+        Objects.requireNonNull(fragments, "fragments is null");
+
+        Processor<?, ?> processor = (Processor<?, ?>)processorGetter.invoke();
+        MethodHandle mh = processor instanceof Linkage linkage
+                ? linkage.linkage(List.of(fragments), type)
+                : defaultProcessMethodHandle(type, processor, List.of(fragments));
+
+        return new ConstantCallSite(mh);
+    }
+
+    /**
+     * Creates a simple {@link StringTemplate} and then invokes the processor's process method.
+     *
+     * @param fragments fragments from string template
+     * @param processor {@link Processor} to process
+     * @param values    array of expression values
+     *
+     * @return result of processing the string template
+     */
+    private static Object defaultProcess(
+            List<String> fragments,
+            Processor<?, ?> processor,
+            Object[] values
+    ) throws Throwable {
+        List<Object> asList = Collections.unmodifiableList(new ArrayList<>(Arrays.asList(values)));
```

Comment:
```suggestion
        List<Object> asList = List.of(values);
```
For defensive copy.
Don't think processors are harmed by the null-hostile behavior of these list, for many template implementations already use null-hostile lists.

Reply:
The values list can't be null-hostile for the same reason that string concatenation can't be null-hostile. Please point to examples of null-hostile lists (other that fragments) being used in the template code. Apologies if I misinterpreted your meaning.

Reply:
Changed